plot_aggregate_loanbooks.R

```diff
@@ -47,762 +53,907 @@ if (file.exists(here::here(".env"))) {
 
 # load required data----
 
-## matched loanbook data----
-matched_prioritized <- readr::read_csv(
-  file.path(input_path_matched, "matched_prioritized_final_for_plots.csv"),
-  col_types = col_types_matched_prio_all_groups,
-  col_select = dplyr::all_of(col_select_matched_prio_all_groups)
-)
+# TODO: check this works appropriately for meta level / by_group = NULL
+# currently plots do not handle by_group == NULL well, as they cannot turn NULL into a symbol
+if (is.null(by_groups)) {
+  file_by_group <- ""
+} else {
+  file_by_group <- glue::glue("_by_{paste(by_groups, collapse = \"_\")}")
+}
 
 ## company level results----
-company_aggregated_alignment_net_tms  <-
-  readr::read_csv(file.path(output_path_aggregated, "company_aggregated_alignment_net_tms.csv"))
-
-company_aggregated_alignment_net_sda <-
-  readr::read_csv(file.path(output_path_aggregated, "company_aggregated_alignment_net_sda.csv"))
+company_aggregated_alignment_net  <-
+  readr::read_csv(
+    file = file.path(
+      output_path_aggregated,
+      glue::glue("company_exposure_net_aggregate_alignment{file_by_group}.csv")
+    )
+  )
 
-company_aggregated_alignment_bo_po_tms <-
-  readr::read_csv(file.path(output_path_aggregated, "company_aggregated_alignment_bo_po_tms.csv"))
+company_aggregated_alignment_bo_po <-
+  readr::read_csv(
+    file = file.path(
+      output_path_aggregated,
+      glue::glue("company_exposure_bo_po_aggregate_alignment{file_by_group}.csv")
+    )
+  )
 
 ## loanbook level results----
 loanbook_exposure_aggregated_alignment_bo_po <-
-  readr::read_csv(file.path(output_path_aggregated, "loanbook_exposure_aggregated_alignment_bo_po.csv"))
+  readr::read_csv(
+    file = file.path(
+      output_path_aggregated,
+      glue::glue("loanbook_exposure_bo_po_aggregate_alignment{file_by_group}.csv")
+    )
+  )
 
 loanbook_exposure_aggregated_alignment_net <-
-  readr::read_csv(file.path(output_path_aggregated, "loanbook_exposure_aggregated_alignment_net.csv"))
+  readr::read_csv(
+    file = file.path(
+      output_path_aggregated,
+      glue::glue("loanbook_exposure_net_aggregate_alignment{file_by_group}.csv")
+    )
+  )
 
-# generate plots for system-wide analysis----
 
+# generate plots for system-wide analysis----
+### sankey plot----
+# TODO: when benchmarks get re-introduced, they need to be removed here
 # Plot sankey plot of financial flows scenario alignment - examples
+if (length(by_groups) <= 1) {
+  if (!is.null(company_aggregated_alignment_net)) {
+    data_sankey_sector <- prep_sankey(
+      company_aggregated_alignment_net,
+      region = "global",
+      year = 2027,
+      group_var = by_groups,
```

Comment:
IIRC the `group_var` argument of `prep_sankey` can only accept a character vector of length `1`, but passing it the parameter `by_groups` implies there could be a situation where more than one grouping variable is passed. Is that a problem?